app/assets/src/components/modules/LabeledSlider.jsx

```diff
@@ -1,17 +1,46 @@
+import PropTypes from "prop-types";
 import React from "react";
 import Slider from "rc-slider";
 
 class LabeledSlider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChangeParent = this.props.onChange;
+
+    this.state = {
+      value: this.props.value
+    };
+
+    this.onChange = this.onChange.bind(this);
+  }
+
+  onChange(newValue) {
+    this.setState({ value: newValue });
+    if (typeof this.onChangeParent == "function") {
+      this.onChangeParent();
```

Comment:
what is the onChangeParent needed for actually?

Reply:
If the parent (in this case the heatmap page) wants to do any else when the LabeledSlider value is changed.

Labeled Slider hooks to the on change of Labeled Slider to update its inner fields.
The SamplesHeatmap hooks to the on change of Labeled Slider to update the heatmap when value changes.